### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by muttmutt** [jcsalterego/Sky.app#30]
I'm seeing high CPU usage with the Sky app. Let me know what additional data you need.
<img width="1359" alt="Screenshot 2024-12-26 at 4 15 22 PM" src="https://github.com/user-attachments/assets/101d2f33-08c8-4ce7-a4a9-cb195ccc862a" />

**Comment by jcsalterego** [comment]
Aw. Probably just the price of running WebKit. If you quit and reload the app, is there a surefire page that makes the CPU go brrrr?

**Comment by muttmutt** [comment]
Yeah, it resets things but then about 3 minutes later it’s eating a CPU again. I’ll test with different pages to see if one is worse than another.

--
Phil Scarr
***@***.***
🦋 Find me on [Bluesky](https://bsky.app/profile/muttmutt.com)!

**Comment by sweetmslily** [comment]
Just wanted to confirm that this is happening for me too. Checking htop in Terminal shows a WebKit process eating CPU, closing Sky makes the process disappear. It takes anywhere between fifteen minutes to a couple hours before the app starts eating the CPU. Only reason I noticed is I keep tabs on the fan/temps in my menu bar and it causes a spike in fan speed and temps on my machine, pushing idle to 80-90 degrees instead of the typical idle range with a bunch of apps running being 55-65 degrees. Let me know if there's anything else I can provide that helps you quash the bug. 

![Image](https://github.com/user-attachments/assets/578f9db8-0eba-45d6-bd75-ee743ff2a99a)

**Comment by jcsalterego** [comment]
Thanks for the info. Unfortunately I don't think the PWA is meant to be a long running process, which is why I personally restart the app quite often. As far as I've observed, React does a lot of DOM manipulation without clear garbage collection. You can see for yourself by opening up the app in Safari/Chrome and observing the Elements tab.

Previous attempts to do some manual garbage collection by manipulating the DOM itself resulted in some nasty crashes.

Unless someone shows up with a patch or a key insight, I somewhat doubt this is in the scope of Sky.app. Thanks!

**Comment by jcsalterego** [comment]
I'm going to go ahead and close this - but do give the latest version a shot and see if it's gotten better.
